lib/ASRouterTargeting.jsm

```diff
@@ -153,7 +153,14 @@ const TargetingGetters = {
 
   // Temporary targeting function for the purposes of running the simplified onboarding experience
   get isInExperimentCohort() {
-    return Services.prefs.getIntPref(ONBOARDING_EXPERIMENT_PREF, 0);
+    const allProviders = Services.prefs.getStringPref(ONBOARDING_MESSAGE_PROVDIER_EXPERIMENT_PREF, "");
+    try {
+      const {cohort} = JSON.parse(allProviders).find(i => i.id === "onboarding");
+      return (typeof cohort === "number" ? cohort : 0);
```

Comment:
we don't _have_ to check that it's a number here, since "1" could still technically be valid, but if we want to keep consistent return types we should do this check. up to you @k88hudson